src/shared/Microsoft.Git.CredentialManager/GitVersion.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Git.CredentialManager
+{
+    public class GitVersion : IComparable, IComparable<GitVersion>
+    {
+        private List<int> components;
+
+        public GitVersion(string versionString)
+        {
+            if (versionString is null)
+            {
+                components = new List<int>();
+                return;
+            }
+
+            string[] splitVersion = versionString.Split('.');
+            components = new List<int>(splitVersion.Length);
+
+            foreach (string part in splitVersion)
+            {
+                if (Int32.TryParse(part, out int component))
+                {
+                    components.Add(component);
+                }
+                else
+                {
+                    // Exit at the first non-integer component
```

Comment:
Makes sense, this would turn `2.33.0.vfs.2.3` into `[2, 33, 0]`.

TBH version comparing is (in my experience) somewhat of an ill-defined problem, and the more simplistic we can make it without breaking our use cases, the better.

As a point of reference, I would like to link [to the version comparison in Git for Windows' installer](https://github.com/git-for-windows/build-extra/blob/0ad275c0edab6a2e0bf36622411417948bca3dcc/installer/install.iss#L890-L953) and [the code in `git update-git-for-windows`](https://github.com/git-for-windows/build-extra/blob/0ad275c0edab6a2e0bf36622411417948bca3dcc/git-extra/git-update-git-for-windows#L11-L44). As you can see, they both do different things, and neither of them are complete.

I also had a brief hunch that there might be something in .NET, and there is: `System.Version`. Alas, it seems that [the only way to parse a string into an instance of that class](https://docs.microsoft.com/en-us/dotnet/api/system.version.-ctor?view=net-5.0#System_Version__ctor_System_String_) is very limited and would choke on any letters (such as `vfs`).
 
All this is to say: I think this implementation is nice and small, and good enough for our purposes. Thank you for implementing it!